### PR TITLE
[RFC] citra-qt: Print stack trace redux

### PR DESCRIPTION
**Pull request by MerryMage**
Original PR: #1874 

Things changed from original PR:

* Rewritten to use `setjmp`/`longjmp` so that Qt can be used.
* Windows-only for now.
* Requires a pdb file to be present for symbols to work on Windows.
* Design of crash dialog TBD, it's just a QMessageBox but can be anything.

Screenshot (real design tbd):

### CODE REVIEW

File: src/common/crash_handler.cpp [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+static jmp_buf unhandled_exception_jmp_buf;
+static std::vector<std::string> unhandled_exception_stack_trace;
+
+static LONG WINAPI MyUnhandledExceptionFilter(_EXCEPTION_POINTERS*);
+static void GetStackTrace(CONTEXT& c);
+
+void CrashHandler(std::function<void()> try_,
+                  std::function<void(const Common::CrashInformation&)> catch_) {
+    unhandled_exception_called = false;
+    SetUnhandledExceptionFilter(MyUnhandledExceptionFilter);
```
Comment (MerryMage) [inline src/common/crash_handler.cpp]: Unset when done.

**Comment by Subv** [comment]
I think instead of showing a dialog, we should dump the relevant information to a file, perhaps a Visual Studio dmp file.

**Comment by FernandoS27** [comment]
Must agree with Subv. A crash report generated with date, version and additional info would be cooler.

Nice work, though.

**Comment by chris062689** [comment]
A crash report would be much more useful attaching to the forums or discord. If it could include the git hash that would be useful too....

**Comment by MerryMage** [comment]
@chris062689 Regarding the crash dialog: I was thinking of having a copy-pastable textbox. git hash and branch name are things that can be added quite easily.

**Comment by Subv** [comment]
As a side note, a dmp file would also contain information about the registers / variables at crash time

**Comment by MerryMage** [comment]
@Subv A minidump would be reasonable starting point, though I think we'd want slightly more information that just the minidump. Not sure how that would work with the gcc builds (but those aren't supported anyway).

**Comment by yuriks** [comment]
Stack traces are pretty easy to do if it's just windows, but if we're going the minidump route then it's probably better to just use something off-the-shelf like Breakpad.

**Comment by MerryMage** [comment]
Breakpad is certainly an option for a library to integrate.

Breakpad does provide the ability to send minidumps to a server and a server-side interface for parsing minidumps. It does not however provide a user-side UI, though this should be rather straightfoward to implement.

Can I consider this approach agreeable with everyone?

**Comment by chris062689** [comment]
I'd prefer user-uploadable minidumps instead of automatically sending them to a server. We don't have the infrastructure setup for that and that may take time.

**Comment by MerryMage** [comment]
Closed in favour of eventual implementation of alternative suggestion(s) above.

**Pull request closed without merging**